input/script_utils.go

```diff
@@ -2095,6 +2130,14 @@ type CommitScriptTree struct {
 	// RevocationLeaf is the leaf used to spend the output with the
 	// revocation key signature.
 	RevocationLeaf txscript.TapLeaf
+
+	// AuxLeaf is an axuillaruy leaf that can be used to extend the base
+	// commitment script tree with new spend paths, or just as extra
+	// commitment space. When present, this leaf will always be in the
+	// left-most or right-most area of the tapscript tree.
+	//
+	// TODO(roasbeeF): need to ensure commitment position...
```

Comment:
Yeah, in that we need to ensure that the aux leaf is in the right-most part of the tree. I think it already achives that though, as in most places we only had a single tapscript leaf. In some places we have two (delay+revocation), but the way they're passed in, since we'd have an odd number of leaves (2 normal, 1 aux), the two get made into a branch, then combined with the dangling leaf at the top level.